goagen/gen_client/generator.go

```diff
@@ -873,11 +889,16 @@ func (c *Client) {{ $funcName }}(resp *http.Response) ({{ decodegotyperef . .All
 `
 
 	pathTmpl = `{{ $funcName := printf "%sPath%s" (goify (printf "%s%s" .Route.Parent.Name (title .Route.Parent.Parent.Name)) true) ((or (and .Index (add .Index 1)) "") | printf "%v") }}{{/*
-*/}}{{ with .Route }}// {{ $funcName }} computes a request path to the {{ .Parent.Name }} action of {{ .Parent.Parent.Name }}.
-func {{ $funcName }}({{ pathParams . }}) string {
-	return fmt.Sprintf({{ printf "%q" (pathTemplate .) }}, {{ pathParamNames . }})
+*/}}// {{ $funcName }} computes a request path to the {{ .Route.Parent.Name }} action of {{ .Route.Parent.Parent.Name }}.
+func {{ $funcName }}({{ pathParams .Route }}) string {
+	var params []interface{}
```

Comment:
It would be good to avoid having to create a slice for this. This method could be called multiple times per request being handled so the extra allocations / execution time could add up. Granted it is more complicated but it's possible and it seems better to take the time to do this right.

I'm thinking instead of using a temporary variable the code could use a known prefix, e.g. "param" and append an index to it. So in the loop it would be `{{ toString "p" (printf "param%d" $i) $param.Attribute }}` where `$i` is declared in the `range` expression (`{{ range $i, $param := .Params }}`). The code rendering the `fmt.Sprintf` would then also use the same range to build the arguments.